### PR TITLE
Implement special functions

### PR DESCRIPTION
**Pull request by not522**
This PR implements following functions.

- cupy.i0
- cupy.sinc
- cupyx.scipy.special.j0
- cupyx.scipy.special.j1
- cupyx.scipy.special.y0
- cupyx.scipy.special.y1
- cupyx.scipy.special.i0
- cupyx.scipy.special.i1

`sinc` and `j1` are requested at #1223.

### CODE REVIEW

**Comment by toslunar** [comment]
Could you test sinc(0)?

**Comment by toslunar** [comment]
Thanks.  The fix looks correct to me but I have a concern: I don't know if sinpi(x) = pi * x as float numbers in cuda for sufficiently small x.
Clipping abs(x) < 1e-20 seems OK by sinc(x) = 1 - O(x^2) and float64's precision.  It seems numpy is relying on sin(y) = y for sufficiently small y = pi * x.
https://github.com/numpy/numpy/blob/v1.14.3/numpy/lib/function_base.py#L3950

**Comment by not522** [comment]
I find that `sinpi(in0) / (M_PI * in0)` sometimes does not return 1 in small x, e.g. 1e-9. I think clipping abs(x) < 1e-9 seems OK because `1-sinc(1e-9)` is smaller than `2e-18`.

File: tests/cupyx_tests/scipy_tests/special_tests/test_bessel.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import unittest
+
+import cupy
+from cupy import testing
+import cupyx.scipy.special
+
+import scipy.special
```
Comment (toslunar) [inline tests/cupyx_tests/scipy_tests/special_tests/test_bessel.py]: Could you ignore `ImportError`?

**Comment by not522** [comment]
I fixed `ImportError` and added fusion tests.

**Comment by toslunar** [comment]
I found some corner-case bugs in numpy's special functions (numpy/numpy#11204, numpy/numpy#11205).  It's ok to me that cupy keeps from reproducing them.

File: tests/cupyx_tests/scipy_tests/special_tests/test_bessel.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        self.check_unary('y1')
+
+    def test_i0(self):
+        self.check_unary('i0')
+
+    def test_i1(self):
+        self.check_unary('i1')
+
+
+@testing.gpu
```
Comment (toslunar) [inline tests/cupyx_tests/scipy_tests/special_tests/test_bessel.py]: This test doesn't require a GPU.

File: tests/cupyx_tests/scipy_tests/special_tests/test_bessel.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+@testing.with_requires('scipy')
+class TestFusionSpecial(unittest.TestCase):
+
+    @testing.for_dtypes(['f', 'd'])
+    @testing.numpy_cupy_allclose(atol=1e-5)
+    def check_unary(self, name, xp, dtype):
+        a = testing.shaped_arange((2, 3), xp, dtype)
+
+        if xp is cupy:
```
Comment (toslunar) [inline tests/cupyx_tests/scipy_tests/special_tests/test_bessel.py]: `cupy.testing.helper` decorators have `sp_name=None` to get `sparse` module.  How about adding `scipy_name` there?

File: tests/cupyx_tests/scipy_tests/special_tests/test_bessel.py (hunk 2 above)
Comment (not522) [reply tests/cupyx_tests/scipy_tests/special_tests/test_bessel.py]: This test seems to require a GPU. Do you mean `TestFusionCPUSpecial`? (It's fixed.)

File: tests/cupyx_tests/scipy_tests/special_tests/test_bessel.py (hunk 3 above)
Comment (not522) [reply tests/cupyx_tests/scipy_tests/special_tests/test_bessel.py]: Sounds nice! I sent PR #1339.

File: tests/cupyx_tests/scipy_tests/special_tests/test_bessel.py (hunk 2 above)
Comment (toslunar) [reply tests/cupyx_tests/scipy_tests/special_tests/test_bessel.py]: Yes.  It seems these comments are at the correct position (just above `TestFusionCPUSpecial`) when I select the commits before the fix: https://github.com/cupy/cupy/pull/1233/files/c6fe974cd9d99ba38d85166dcc2fa3397010210d
I should have commented on the next line 🤔

**Comment by toslunar** [comment]
Jenkins, test this please.
